source/Personality.cpp

```diff
@@ -469,19 +493,19 @@ const Point &Personality::Confusion() const
 
 void Personality::UpdateConfusion(bool isFiring)
 {
-	// If you're firing weapons, aiming accuracy should slowly improve until it
-	// is 4 times more precise than it initially was.
-	aimMultiplier = .99 * aimMultiplier + .01 * (isFiring ? .5 : 2.);
-
-	// Try to correct for any error in the aim, but constantly introduce new
-	// error and overcompensation so it oscillates around the origin. Apply
-	// damping to the position and velocity to avoid extreme outliers, though.
-	if(confusion.X() || confusion.Y())
-		confusionVelocity -= .001 * confusion.Unit();
-	confusionVelocity += .001 * Angle::Random().Unit();
-	confusionVelocity *= .999;
-	confusion += confusionVelocity * (confusionMultiplier * aimMultiplier);
-	confusion *= .9999;
+	tick++;
```

Comment:
No need to run any numbers if it all collapses to zero.
```suggestion
	if(!confusionMultiplier)
		return
	tick++;
```